### PR TITLE
[kata/apps-cli/M005/S03] Harden CLI CI test suite

### PR DESCRIPTION
**Pull request by gannonh**
## What Changed
See slice plan: S03: Harden CLI CI test suite

## Must-Haves
- R003 (owned): Linear integration tests skip cleanly when `LINEAR_API_KEY` is absent (no false failures)
- R003 (owned): Smoke tests pass with realistic timeouts — no Bun 5s default timeout flakes on subprocess-heavy tests
- Existing test assertions preserved exactly — only restructure, don't weaken coverage
- `npx tsc --noEmit` passes after changes

## Tasks
- T01: Fix linear-config.integration.test.ts skip gating and done-callback timeout
- T02: Add realistic timeouts to slow smoke tests

## Summary by CodeRabbit

* **Tests**
  * Restructured integration tests for improved organization and error handling
  * Extended test timeout configurations to enhance reliability during longer-running test operations

<h3>Greptile Summary</h3>

This PR hardens the CLI CI test suite by fixing two classes of flake: spurious failures when `LINEAR_API_KEY` is absent, and Bun's default 5-second timeout firing on subprocess-heavy smoke tests.

- **`linear-config.integration.test.ts`** — Two top-level tests each carrying a `{ skip: … }` option are collapsed into a single `describe` block. An `if (!API_KEY) { test.skip(…); return; }` guard at describe-scope prevents the real tests from being registered at all when the env var is missing (rather than skipping them after collection). The previous `async (t) =>` signature on the second test was also removed; it called `t.skip()` internally without ever calling `t.done()`, which caused Bun to wait for a done-callback that never fired.
- **`app-smoke.test.ts`** — Four tests that run `npm run build`, `npm pack`, `npm install`, and spawn long-lived child processes are each given an explicit `{ timeout: 60_000 }` option, replacing Bun's 5 s default.

The overall approach is sound. One minor concern: two of the four timed tests ("tarball installs and kata-cli binary resolves" and "kata startup installs pi-mcp-adapter") chain a full TypeScript build with a subsequent `npm install` or network-touching subprocess under the same 60-second budget. On cold CI runners this combined wall-clock time can exceed 60 seconds, meaning those two tests may still occasionally flake — a ceiling of 120 seconds would give a wider safety margin.

<h3>Confidence Score: 4/5</h3>

- Safe to merge — changes are test-only, logic is correct, and the skip-gating fix is a clear improvement over the previous done-callback approach.
- Both files are test infrastructure only (no production code touched). The `describe` + early-`return` skip pattern is idiomatic and correct in Bun/Jest/Vitest. The timeout additions are straightforwardly better than the 5 s default. The only residual risk is that 60 s may still be too tight for the build+install tests on cold CI runners, which could reintroduce flakes at a lower frequency.
- `apps/cli/src/tests/app-smoke.test.ts` lines 396 and 539 — the two tests that chain a full build with a network install may benefit from a 120 s timeout.

<details><summary><h3>Important Files Changed</h3></summary>

| Filename | Overview |
|----------|----------|
| apps/cli/src/resources/extensions/kata/tests/linear-config.integration.test.ts | Restructured from two standalone tests with inline skip options to a single `describe` block with early `return` when `LINEAR_API_KEY` is absent; also removes the `async (t)` done-callback signature that caused Bun timeout issues. Logic is correct and the pattern is idiomatic, though the "no projects" branch now silently passes instead of showing as skipped. |
| apps/cli/src/tests/app-smoke.test.ts | Adds `{ timeout: 60_000 }` to four subprocess-heavy tests; a meaningful improvement over Bun's 5 s default. Tests that combine a full `npm run build` with a subsequent `npm install` (tests 7 and 8) may still flake under cold CI conditions — a higher ceiling (120 s) would be safer for those two. |

</details>

</details>

<h3>Flowchart</h3>

```mermaid
%%{init: {'theme': 'neutral'}}%%
flowchart TD
    A[Test runner collects describe block] --> B{LINEAR_API_KEY set?}
    B -- No --> C[register test.skip placeholder]
    C --> D[return — no real tests registered]
    D --> E[CI: 1 skipped, 0 failures]
    B -- Yes --> F[register 'resolves team' test]
    F --> G[register 'resolves project' test]
    G --> H[run both tests against Live Linear API]
    H --> I{projects.length == 0?}
    I -- Yes --> J[return early — test passes silently]
    I -- No --> K[assert resolved project fields]
    K --> L[CI: 2 passed]
```

<details><summary><h3>Comments Outside Diff (1)</h3></summary>

1. `apps/cli/src/tests/app-smoke.test.ts`, line 396-458 ([link](https://github.com/gannonh/kata/blob/8f6c3299b8b14384587588eeed1ae9cf3d78a3e1/apps/cli/src/tests/app-smoke.test.ts#L396-L458)) 

   <a href="#">P2</a> **60 s budget may be tight for build + remote install**

   This test runs two potentially slow operations under the same 60-second window:

   1. `execSync("npm run build", …)` — TypeScript compilation, bundling, etc.
   2. `execSync(\`npm install --prefix ${tmp} ${tarballPath} …\`, …)` — package download + postinstall scripts.

   On a cold CI runner (no cache) or with a slow npm registry, the combined wall-clock time can easily exceed 60 seconds, meaning this test could still flake. The sibling test "kata startup installs pi-mcp-adapter" has the same profile (build + subprocess that installs over the network).

   Consider bumping the timeout for these two network-touching tests:

   

   <details><summary>Prompt To Fix With AI</summary>

   `````markdown
   This is a comment left during a code review.
   Path: apps/cli/src/tests/app-smoke.test.ts
   Line: 396-458

   Comment:
   **60 s budget may be tight for build + remote install**

   This test runs two potentially slow operations under the same 60-second window:

   1. `execSync("npm run build", …)` — TypeScript compilation, bundling, etc.
   2. `execSync(\`npm install --prefix ${tmp} ${tarballPath} …\`, …)` — package download + postinstall scripts.

   On a cold CI runner (no cache) or with a slow npm registry, the combined wall-clock time can easily exceed 60 seconds, meaning this test could still flake. The sibling test "kata startup installs pi-mcp-adapter" has the same profile (build + subprocess that installs over the network).

   Consider bumping the timeout for these two network-touching tests:

   

   How can I resolve this? If you propose a fix, please make it concise.
   `````
   </details>

</details>

<details><summary>Prompt To Fix All With AI</summary>

`````markdown
This is a comment left during a code review.
Path: apps/cli/src/tests/app-smoke.test.ts
Line: 396-458

Comment:
**60 s budget may be tight for build + remote install**

This test runs two potentially slow operations under the same 60-second window:

1. `execSync("npm run build", …)` — TypeScript compilation, bundling, etc.
2. `execSync(\`npm install --prefix ${tmp} ${tarballPath} …\`, …)` — package download + postinstall scripts.

On a cold CI runner (no cache) or with a slow npm registry, the combined wall-clock time can easily exceed 60 seconds, meaning this test could still flake. The sibling test "kata startup installs pi-mcp-adapter" has the same profile (build + subprocess that installs over the network).

Consider bumping the timeout for these two network-touching tests:

```suggestion
test("tarball installs and kata-cli binary resolves", { timeout: 120_000 }, async () => {
```

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/cli/src/tests/app-smoke.test.ts
Line: 539

Comment:
**Same build + network install concern here**

This test runs `npm run build` then spawns a process that installs `pi-mcp-adapter` globally (via `npm_config_prefix`). The internal kill timer fires at 20 s, but the build itself can approach or exceed the remaining budget in that window. Using `120_000` here (consistent with the suggestion above) gives a comfortable margin.

```suggestion
test("kata startup installs pi-mcp-adapter into an isolated npm prefix", { timeout: 120_000 }, async () => {
```

How can I resolve this? If you propose a fix, please make it concise.

---

This is a comment left during a code review.
Path: apps/cli/src/resources/extensions/kata/tests/linear-config.integration.test.ts
Line: 74-77

Comment:
**Silent `return` changes CI visibility from "skip" to "pass"**

Previously this branch called `t.skip(...)` (which is why the old signature had `async (t) =>`), causing the test to be reported as **skipped** in CI output. The new code just `return`s early — the test runner will record it as **passed** instead.

This is probably intentional given the "skip silently" comment, but it does mean CI will no longer hint that the project-validation path was never exercised when no Linear projects exist in the account. Worth a conscious acknowledgment (or a `console.warn`/log) if observability of the untested branch matters.

How can I resolve this? If you propose a fix, please make it concise.
`````

</details>

<sub>Last reviewed commit: ["feat(S03/T02): add 6..."](https://github.com/gannonh/kata/commit/8f6c3299b8b14384587588eeed1ae9cf3d78a3e1)</sub>

> Greptile also left **2 inline comments** on this PR.

### CODE REVIEW

File: apps/cli/src/tests/app-smoke.test.ts [hunk 1]
```diff
@@ -309,7 +309,7 @@ test("loadStoredEnvKeys does not overwrite existing env vars", async () => {
 // 6. npm pack produces valid tarball with correct file layout
 // ═══════════════════════════════════════════════════════════════════════════
 
-test("npm pack produces tarball with required files", async () => {
+test("npm pack produces tarball with required files", { timeout: 60_000 }, async () => {
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/tests/app-smoke.test.ts:312]: This is incorrect — Bun 1.3.8 supports the `test(name, options, fn)` signature with `{ timeout }` in the options object. All 13 smoke tests pass with this syntax (verified locally). The `test(name, fn, timeout)` positional form is not the only supported API.

File: apps/cli/src/tests/app-smoke.test.ts [hunk 2]
```diff
@@ -536,7 +536,7 @@ test("loader.ts injects --mcp-config into process.argv", () => {
   );
 });
 
-test("kata startup installs pi-mcp-adapter into an isolated npm prefix", async () => {
+test("kata startup installs pi-mcp-adapter into an isolated npm prefix", { timeout: 60_000 }, async () => {
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/tests/app-smoke.test.ts:539]: The test completes in ~21s, so 60s provides ~3x margin which is sufficient. The task plan specifies `60_000` and this matches the other three slow tests for consistency. If CI environments prove slower we can revisit, but 60s is a realistic ceiling for now.

File: apps/cli/src/resources/extensions/kata/tests/linear-config.integration.test.ts [hunk 3]
```diff
@@ -59,23 +62,17 @@ test(
...
 
-    // Fetch projects scoped to the first team so team and project are guaranteed
-    // to be associated (avoids a test that passes with incompatible config).
     const projects = await client.listProjects({ teamId: teams[0].id, first: 25 });
 
     if (projects.length === 0) {
-      t.skip("No Linear projects available for the first team in integration validation");
+      // No projects available — nothing to validate; skip silently
       return;
     }
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/resources/extensions/kata/tests/linear-config.integration.test.ts:77]: Intentional. The old `t.skip()` pattern required the `async (t) =>` callback signature, which triggered Bun's done-callback timeout bug (the root cause T01 fixed). This `return` is only reached when `LINEAR_API_KEY` IS set but the first team has no projects — an uncommon edge case within an already-gated integration test. A silent pass is correct here: there's nothing to validate.

File: apps/cli/src/resources/extensions/kata/tests/linear-config.integration.test.ts [hunk 4]
```diff
@@ -59,23 +62,17 @@ test(
...
     assert.ok(teams.length > 0, "expected at least one Linear team");
 
-    // Fetch projects scoped to the first team so team and project are guaranteed
-    // to be associated (avoids a test that passes with incompatible config).
     const projects = await client.listProjects({ teamId: teams[0].id, first: 25 });
 
     if (projects.length === 0) {
-      t.skip("No Linear projects available for the first team in integration validation");
+      // No projects available — nothing to validate; skip silently
       return;
```
[the remark this answers is not in the document]
Comment (gannonh) [reply apps/cli/src/resources/extensions/kata/tests/linear-config.integration.test.ts:76]: Same as above — the `return` replacing `t.skip()` is intentional. The `async (t) =>` signature required by `t.skip()` was the root cause of the 5s done-callback timeout bug that this PR fixes. The "no projects on first team" branch is an uncommon edge case within an integration test that already skips entirely without `LINEAR_API_KEY`. No regression risk here.
